source/Calamari/Kubernetes/KubernetesContextScriptWrapper.cs

```diff
@@ -421,6 +425,79 @@ void SetupContextForAmazonServiceAccount(string @namespace, string clusterUrl, s
                 var clusterName = variables.Get(SpecialVariables.EksClusterName);
                 log.Info($"Creating kubectl context to {clusterUrl} (namespace {@namespace}) using EKS cluster name {clusterName}");
 
+                log.Verbose("Attempting to authenticate with aws-cli");
+                if (TrySetAws())
+                {
+                    try
+                    {
+                        var region = GetEksClusterRegion(clusterUrl);
+                        if (string.IsNullOrWhiteSpace(region))
+                        {
+                            log.Verbose("The EKS cluster Url specified should contain a valid aws region name");
+                        }
+                        
+                        var awsCliVersion = GetAwsCliVersion();
+
+                        if (new Version(awsCliVersion).CompareTo(new Version("1.16.156")) > 0)
```

Comment:
A few things I'm wondering about this `Version` usage:
* Could the `GetAwsCliVersion` method directly return a `Version` rather than a `string`?
* Could we extract the comparison Version to a named field/constant, so that it's got a name, something like `minmumAwsCliVersionForAuth`?
* Is it possible to just directly use comparison operators rather than the `CompareTo` method? Then it could look more like:
    ```
    if (awsCliVersion > minimumAwsCliVersionForAuth)
    ```